src/modelmanager.cpp

```diff
@@ -698,18 +699,39 @@ Status ModelManager::reloadModelWithVersions(ModelConfig& config) {
             return StatusCode::OK;
         }
     }
+
     getVersionsToChange(config, model->getModelVersions(), requestedVersions, versionsToStart, versionsToReload, versionsToRetire);
 
-    if (versionsToStart->size() > 0) {
-        auto blocking_status = addModelVersions(model, fs, config, versionsToStart);
+    while (versionsToStart->size() > 0) {
```

Comment:
Why are we using while here? Is second iteration possible here?
If it's just to check condition maybe we don't need a loop

Reply:
multiple loops are possible. In situation when several latest versions will be invalid, there will be attempt to add the latests only. When it fails, ovms need to reassess the version policy excluding the failed latest version. It means  the second loop which can still include corrupted version. There could be as many loops as many corrupted versions get detected. We should continue the loop till there are no more versions to add (all corrupted) or the load is successful.